### ISSUE TITLE
consensus: Introduce block "Consensus State" labels

### ISSUE THREAD

**Issue by fed-franz** [dusk-network/rusk#1158]
#### Summary

Introduce a "label" for blocks to indicate their state with respect to their consensus status.
Define the following labels:
 - *Accepted*:  the block has been accepted by the node (i.e., it reached a quorum of votes at a certain iteration)
 - *Attested*: the block has been accepted and it has a valid Attestation (see the definition below) for all iterations from 1 to the one reaching the Quorum
 - *Final*: the block has been accepted, attested, and all previous blocks are also _final_

#### Detailed Description

*Attestation*
An Attestation is a collection of votes along with the bitset of the voters. It is basically what we used to call a Certificate, but it used for both Agreement Quorums and Nil Quorums. 

Optionally, we might allows Attestations to be incomplete, i.e. to contain less-then-quorum votes, although these won't count for Attesting a block. To avoid ambiguities, we might define an Attestation to be any collection of valid votes for the block candidate, and a Quorum Attestation to be a collection reaching an Agreement or Nil Quorum.

Additionally, in case of Nil Quorum, we might also include any positive vote, which can then account for the _Rolling Finality_ feature.

#### Relevant Context

The introduction Quorum Attestations (and Attested blocks) is intended to distinguish the case in which a block cannot be replaced by a lower-iteration block with the same predecessor (which is what we used to define as 'final') from the case in which the block cannot be replaced by ANY block (including a block at the same height but on a different fork) which what we now consider as final.

In other words:
 - an Accepted block has reached an agreement (i.e., it has an Agreement Attestation) but not at the first iteration and does not include a Nil Attestation for all failed iterations
 - an Attested block has both the Agreement Attestation and a Nil Attestation for all failed iterations; this means that it cannot be replaced by another block with the same parent (but it could be replaced by a fork)
 - a Final block has all Attestations and its parent is also labeled as Final

#### Possible Solution

Each block has a "Consensus State" (Accepted/Attested/Final).
The state can be implicitly deduced from the Attestations it includes or we can add an explicit label to the internal block structure.
